spring-boot-security-taglibs/src/test/java/org/baeldung/security/HomeControllerTest.java

```diff
@@ -0,0 +1,42 @@
+package org.baeldung.security;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+public class HomeControllerTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Test
+    public void home() throws Exception {
```

Comment:
Tests on baeldung should follow the naming convention givenXXX_whenYYY_thenZZZ or whenYYY_thenZZZ.

Also, please let's have a different test method for each. You are actually already doing this because of your comment above each assertion. Instead you can achieve both by doing:

```java
@Test
public void whenUserIsAuthenticatedThenAuthenticatedSectionsShowOnSite() {}
```

etc.

Also, please demonstrate what happens when they are not authenticated.